File: cpp/Ice/callback/CallbackSenderI.cpp
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

#include "CallbackSenderI.h"
#include <Ice/Ice.h>
#include <iostream>

using namespace std;
using namespace Demo;

void
CallbackSenderI::initiateCallback(optional<CallbackReceiverPrx> proxy, const Ice::Current&)
{
    cout << "initiating callback" << endl;
    try
    {
        proxy->callback();
    }
    catch (const Ice::Exception& ex)
    {
        cout << ex << endl;
    }
}

void
CallbackSenderI::shutdown(const Ice::Current& c)
{
    cout << "shutting down..." << endl;
    c.adapter->getCommunicator()->shutdown();
}

File: java/Ice/callback/CallbackSenderI.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

import com.zeroc.demos.Ice.callback.Demo.*;

public final class CallbackSenderI implements CallbackSender
{
    @Override
    public void initiateCallback(CallbackReceiverPrx proxy, com.zeroc.Ice.Current current)
    {
        System.out.println("initiating callback");
        try
        {
            proxy.callback();
        }
        catch(com.zeroc.Ice.LocalException ex)
        {
            ex.printStackTrace();
        }
    }

    @Override
    public void shutdown(com.zeroc.Ice.Current current)
    {
        System.out.println("Shutting down...");
        try
        {
            current.adapter.getCommunicator().shutdown();
        }
        catch(com.zeroc.Ice.LocalException ex)
        {
            ex.printStackTrace();
        }
    }
}
